opslib/osm/config/mysql.py

```diff
@@ -0,0 +1,50 @@
+# pylint: disable=E0213
+import re
+from typing import Optional
+
+
+from ..validator import ModelValidator, validator
+
+r = r"^mysql\:\/\/{}@{}/{}?$".format(
+    r"(?P<username>[_\w]+):(?P<password>[\w\W]+)",
+    r"(?P<host>[\.\w]+):(?P<port>\d+)",
+    r"(?P<database>[_\w]+)",
+)
+
+
```

Comment:
This could be stored inside the class, no? Do we need to run the regex every time we access a property?
I know that performance isn't an issue but compiling the regex every time we access the property it seems a bit much